File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    
    q_len = len(queue1)
    q1 = deque(queue1)
    q2 = deque(queue2)
    q1_sum = sum(queue1)
    q2_sum = sum(queue2)
    
    if (q1_sum + q2_sum) % 2 == 1:
        return -1
    
    cnt = 0
    task = 0
    while cnt < q_len * 4:
        if q1_sum > q2_sum:
            q1_num = q1.popleft()
            q2.append(q1_num)
            q1_sum -= q1_num
            q2_sum += q1_num
            task += 1
        elif q2_sum > q1_sum:
            q2_num = q2.popleft()
            q1.append(q2_num)
            q2_sum -= q2_num
            q1_sum += q2_num
            task += 1
        else:
            return task
        cnt += 1
    
    return -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        long q1Sum = Arrays.stream(queue1).sum();
        long q2Sum = Arrays.stream(queue2).sum();
        
        var q1 = new ArrayDeque<Long>();
        var q2 = new ArrayDeque<Long>();
        
        for (int i = 0; i < queue1.length; i++) {
            q1.addLast((long)queue1[i]);
            q2.addLast((long)queue2[i]);
        }
        
        var count = 0;
        while (count != queue1.length * 4) {
            if (q1Sum == q2Sum) return count;
            
            if (q1Sum > q2Sum) {
                // q1에서 pop
                var e = q1.removeFirst();
                q2.addLast(e);
                q1Sum -= e;
                q2Sum += e;
            } else {
                // q2에서 pop
                var e = q2.removeFirst();
                q1.addLast(e);
                q1Sum += e;
                q2Sum -= e;
            }
            
            count += 1;
        }
        
        return -1;
    }
}